src/ol/interaction/Snap.js

```diff
@@ -306,6 +307,8 @@ class Snap extends PointerInteraction {
           segment: result.segment,
         }),
       );
+    } else {
+      this.dispatchEvent(new UnsnapEvent());
```

Comment:
It looks like this will dispatch an unsnap event on every event that the interaction handles (unless snapped).  So as you move your mouse around the map, with every mousemove, there will be an unsnap event.  Ideally this event would only be dispatched on the single event where the state changes from snapped to unsnapped.